source/common/stream_info/utility.cc

```diff
@@ -34,6 +35,11 @@ const std::string ResponseFlagUtils::toShortString(const StreamInfo& stream_info
 
   static_assert(ResponseFlag::LastFlag == 0x2000, "A flag has been added. Fix this code.");
 
+  // stream_info.protocol() is only available for HTTP requests.
+  if (stream_info.protocol() && !stream_info.hasAnyResponseFlag() && !stream_info.responseCode()) {
```

Comment:
Nit: prefer `stream_info.protocol() != nulptr`

Reply:
`protocol` is a `absl::optional` instance, so won't be able to use `!= nullptr`.